### ISSUE TITLE
apt module: do not use aptitude

### ISSUE THREAD

**Issue by merifri** [ansible/ansible#16884]
##### ISSUE TYPE
- Bug Report / Feature Idea
##### ANSIBLE VERSION

``` console
$ ansible --version
ansible 2.1.0.0
```
##### OS / ENVIRONMENT

Ubuntu 16.04.1
##### SUMMARY

Ubuntu 16 comes without `aptitude` package:

``` console
 "Could not find aptitude. Please ensure it is installed."
```
##### STEPS TO REPRODUCE

Tasks with `upgrade` parameter fails: 

``` console
~/ansible-scripts $ ansible -i inventory/hosts lab -m apt -a "upgrade=full"
lab.xxx | FAILED! => {
    "changed": false, 
    "failed": true, 
    "msg": "Could not find aptitude. Please ensure it is installed."
}
~/ansible-scripts $ ansible -i inventory/hosts lab -m apt -a "upgrade=yes"
lab.xxx | FAILED! => {
    "changed": false, 
    "failed": true, 
    "msg": "Could not find aptitude. Please ensure it is installed."
}
~/ansible-scripts $ ansible -i inventory/hosts lab -m apt -a "upgrade=full"
lab.xxx | FAILED! => {
    "changed": false, 
    "failed": true, 
    "msg": "Could not find aptitude. Please ensure it is installed."
}
~/ansible-scripts $ ansible -i inventory/hosts lab -m apt -a "upgrade=safe"
lab.xxx | FAILED! => {
    "changed": false, 
    "failed": true, 
    "msg": "Could not find aptitude. Please ensure it is installed."
}
```

Only `upgrade=dist` work fine:

``` console
~/ansible-scripts $ ansible -i inventory/hosts lab -m apt -a "upgrade=dist"
lab.xxx | SUCCESS => {
    "changed": false, 
    "msg": "Reading package lists...\nBuilding dependency tree...\nReading state information...\nCalculating upgrade...\n0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.\n", 
    "stderr": "", 
    "stdout": "Reading package lists...\nBuilding dependency tree...\nReading state information...\nCalculating upgrade...\n0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.\n", 
    "stdout_lines": [
        "Reading package lists...", 
        "Building dependency tree...", 
        "Reading state information...", 
        "Calculating upgrade...", 
        "0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded."
    ]
}
```

**Comment by ghost** [comment]
A better choice would be apt-get. Neither do Debian 7/8 minimals come with aptitude.

I don't recommend using "apt" that seems to be getting popular lately.

Debian 6/Squeeze, even though EOL, doesn't recognize apt.

Eg:

```
Spawning container squeeze on /home/user/.containers/debian/old-debian-releases/squeeze.
Press ^] three times within 1s to kill container.
/etc/localtime is not a symlink, not updating container timezone.
root@squeeze:~# apt update
-bash: apt: command not found
root@squeeze:~# apt-get update
Hit http://archive.debian.org squeeze Release.gpg
Ign http://archive.debian.org/debian/ squeeze/main Translation-en
Hit http://archive.debian.org squeeze-lts Release.gpg
Ign http://archive.debian.org/debian/ squeeze-lts/main Translation-en
Hit http://archive.debian.org squeeze Release
Hit http://archive.debian.org squeeze-lts Release
Hit http://archive.debian.org squeeze/main i386 Packages
Ign http://archive.debian.org squeeze-lts/main i386 Packages/DiffIndex
Hit http://archive.debian.org squeeze-lts/main i386 Packages
Reading package lists... Done
root@squeeze:~#
```

**Comment by alikins** [comment]
This appears to be a submission about a module ('apt' from ansible-modules-core).

We would appreciate if you can submit this there (https://github.com/ansible/ansible-modules-core)
instead. (If we move it, the owner of the issue resets to ansible instead of you).

**Comment by YPCrumble** [comment]
@Merifri did you add this as an issue yet? Having the same problem, happy to add but don't want to create a duplicate.

**Comment by merifri** [comment]
@YPCrumble no, I found similar issue: https://github.com/ansible/ansible-modules-core/issues/3523

**Comment by mamoit** [comment]
The issue ansible/ansible-modules-core#3523 has been closed since the bug tracking moved here.
This issue should be re-opened @alikins.

**Comment by dev169** [comment]
@mamoit @alikins I think this should be marked a duplicate of https://github.com/ansible/ansible/issues/18987 (which was migrated from https://github.com/ansible/ansible-modules-core/issues/3523). I will continue discussion there.

**Comment by pharazon** [comment]
Lol makes me think what is the benefit of ansible over a bash script. I make a simple ansible playbook to upgrade my servers, but it turns out ansible core is broken and doesn't support Ubuntu (or Debian).

**Comment by michaelBenin** [comment]
Is this fixed in 2.2.0?

**Comment by merifri** [comment]
@michaelBenin No. Issue with apt module: https://github.com/ansible/ansible/issues/18987

**Comment by sonjz** [comment]
based on: https://github.com/ansible/ansible-modules-core/issues/3523#issuecomment-222368182

Added this early in the bootstrap to reenable apt (you can remove the becomes if you do that elsewhere)
```
- name: "FIX: Ubuntu 16.04 LTS doesn't come with certain modules, required by ansible"
  raw: apt-get install python-minimal aptitude -y
  become: true
  become_user: root
  become_method: sudo
```
